### PR TITLE
Remove IMetricsManager and use SDK logging hook.

### PR DESCRIPTION
**Pull request by MikeStall**

### CODE REVIEW

**Comment by mathewc** [comment]
Let's hold off on refactoring any of the metrics pipeline until we finish the metrics work we're doing.

**Comment by mathewc** [comment]
Closing this since it is out of date. Currently IMetricsLogger is used for things other than function started/completed. Also the ScriptHost itself is an event producer - not all events are coming out of the core SDK.

Feel free to open another PR or issue if you still think there is something to do here.

**Comment by mathewc** [comment]
Does this PR compile? E.g. ScriptHostManager calls IMetricsLogger.LogEvent which is on the interface you removed, but I don't see any changes to that file.

File: src/WebJobs.Script/Description/CSharp/CSharpFunctionInvoker.cs [hunk 1]
```diff
@@ -216,24 +210,9 @@ private void RestorePackages()
             }
             catch
             {
-                if (startedEvent != null)
-                {
-                    startedEvent.Success = false;
-                    TraceWriter.Verbose(string.Format("Function completed (Failure, Id={0})", invocationId));
```
Comment (mathewc) [inline src/WebJobs.Script/Description/CSharp/CSharpFunctionInvoker.cs]: You're removing logging statements we need to retain

**Comment by MikeStall** [comment]
It did at that time. It needs to be rebased.

**Pull request closed without merging**